pkg/configuration/new_fetcher.go

```diff
@@ -4,33 +4,23 @@ import (
 	"log"
 	"net/http"
 
+	remoteasset "github.com/bazelbuild/remote-apis/build/bazel/remote/asset/v1"
 	"github.com/buildbarn/bb-remote-asset/pkg/fetch"
 	pb "github.com/buildbarn/bb-remote-asset/pkg/proto/configuration/bb_remote_asset/fetch"
-	"github.com/buildbarn/bb-remote-asset/pkg/storage"
-	blobstore_configuration "github.com/buildbarn/bb-storage/pkg/blobstore/configuration"
+	"github.com/buildbarn/bb-storage/pkg/blobstore"
 	bb_digest "github.com/buildbarn/bb-storage/pkg/digest"
 
-	remoteasset "github.com/bazelbuild/remote-apis/build/bazel/remote/asset/v1"
-
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
-// NewFetcherFromConfiguration creates a new Remote Asset API Fetch server from
-// a jsonnet configuration.
+// NewFetcherFromConfiguration creates a new Remote Asset API Fetch
+// server from a jsonnet configuration.
 func NewFetcherFromConfiguration(configuration *pb.FetcherConfiguration,
-	assetStore *storage.AssetStore,
-	casBlobAccessCreator blobstore_configuration.BlobAccessCreator) (remoteasset.FetchServer, error) {
-	var fetcher remoteasset.FetchServer
+	contentAddressableStorage, actionCache blobstore.BlobAccess,
+	pushServer remoteasset.PushServer, maximumSizeBytes int) (fetch.Fetcher, error) {
```

Comment:
We should be careful to avoid counting metrics on Fetch'd assets subsequently Push'd into the push server (once #12 is merged), That sort of double counting would be misleading, and I can definitely see a future version of myself saying "why on earth are we getting pushes?!"